src/nvim/regexp_nfa.c

```diff
@@ -1499,7 +1500,8 @@ static int nfa_regatom(void)
         c = getchr();
       while (ascii_isdigit(c)) {
         if (n > (INT32_MAX - (c - '0')) / 10) {
```

Comment:
8.2.0892 adds a temporary unsigned variable for overflow check but the current check should be sufficient because `n` has `int64_t` type.